tests/unit/test_tracing.py

```diff
@@ -0,0 +1,232 @@
+# Copyright Splunk Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Dict, Optional
+from unittest import TestCase
+
+from opentelemetry.exporter.jaeger.thrift import JaegerExporter
+from opentelemetry.exporter.otlp.proto.grpc.trace_exporter import OTLPSpanExporter
+from opentelemetry.instrumentation.propagators import get_global_response_propagator
+from opentelemetry.sdk.trace.export import ConsoleSpanExporter
+from opentelemetry.sdk.trace.export.in_memory_span_exporter import InMemorySpanExporter
+
+from splunk_otel.env import _EnvVarsABC
+from splunk_otel.propagators import _ServerTimingResponsePropagator
+from splunk_otel.tracing import _do_start_tracing
+
+
+class TestTracing(TestCase):
+    def test_do_start_with_tracing_disabled(self):
+        env = _FakeEnvVars({"OTEL_TRACE_ENABLED": False})
+        res = _do_start_tracing(env)
+        self.assertIsNone(res)
+
+    def test_do_start_tracing_with_defaults(self):
+        env = _FakeEnvVars()
+        tracer_provider = _do_start_tracing(env)
+        self.assertIsNotNone(tracer_provider)
+        env_vars = env.get_env_vars_written()
+        expected_written = {
+            "OTEL_ATTRIBUTE_COUNT_LIMIT": "",
```

Comment:
By default, `environ` will return `None` if the variable does not exist in the environment.

Reply:
Does what environ returns apply here? This is just a representation of the values that have been written to the environment by the production code, which has written explicit empty strings for these values.